### DISCUSSION TITLE
Feature Request: Time based eligibility

### DISCUSSION THREAD

**Discussion by Oatelaus** [aws-samples/iam-identity-center-team#277] [Ideas]
I have searched for other requests on this topic and I could not find any.

I would like to make eligibility policies only active during certain periods of the day - for example I would like to make some eligibility policies available during working hours and then out of hours I would like to provide alternative policies.

I believe there may even be the argument to make eligibility policies active based on a set of times, for example you may want to allow people access for 3.5 hours, 1 hour gap, 3.5 hours.

I believe to achieve this, the following changes would be needed:

- Timezone added into the settings; UTC by default?
- Eligibility policy would need to accept an optional start/end time, preferably an array.
- Future requests would start at the next start time of the eligibility policy you select.
- Requests made which would overlap the end time of the eligibility policy should be timed to end at the end date - perhaps this should be optional or configurable?

I would be happy to contribute to this project and add this functionality.

**Comment by tawoyinfa** [comment]
@Oatelaus contributions are welcomed
